ui/components/app/toast-master/selectors.ts

```diff
@@ -62,25 +63,40 @@ export function selectShowPrivacyPolicyToast(state: State): {
     newPrivacyPolicyToastShownDate,
     onboardingDate,
   } = state.metamask || {};
-  const newPrivacyPolicyDate = new Date(PRIVACY_POLICY_DATE);
-  const currentDate = new Date(Date.now());
+
+  const remoteFeatureFlags = state.metamask?.remoteFeatureFlags || {};
+
+  const policyUpdateDate = String(
+    remoteFeatureFlags.transactionsPrivacyPolicyUpdate || '',
+  );
+
+  // If the feature flag isn't set or is empty, don't show the toast
+  if (!policyUpdateDate) {
+    // No policy update date found
+    return { showPrivacyPolicyToast: false, newPrivacyPolicyToastShownDate };
+  }
+
+  // Create dates and ensure consistent timezone handling
+  const newPrivacyPolicyDate = new Date(policyUpdateDate);
```

Comment:
`newPrivacyPolicyDate` should be validated, as a misconfiguration of this value could result in some strange behavior.

Reply:
Absolutely. Making this change.

Reply:
Added some validation for the feature flag's `newPrivacyPolicyDate`
Prevent showing toast if date is fat fingered or missing
Toast behavior should be consistent with valid  dates